### PR TITLE
chore(): update to Beta.6, including having our own typings.

### PR DESCRIPTION
**Pull request by hansl**
This took longer because of ts2dart which was trying to find stuff in the wrong place.

### CODE REVIEW

**Comment by hansl** [comment]
@jelbourn: please review.

**Comment by hansl** [comment]
The build system is not in good shape, but this is the best I can do considering.

File: ember-cli-build.js [hunk 1]
```diff
@@ -23,11 +24,13 @@ module.exports = function(defaults) {
   var demoAppCssTree = new BroccoliSass(['src/demo-app'], './demo-app.scss', 'demo-app/demo-app.css');
   var demoCssTree = getCssTree('demo-app');
   var componentCssTree = getCssTree('components');
-  var angularAppTree = new Angular2App(defaults);
+  //var angularAppTree = getAppTree('src');
```
Comment (jelbourn) [inline ember-cli-build.js]: Commented-out line?

File: test/karma-test-shim.js [hunk 2]
```diff
@@ -41,9 +41,14 @@ System.config({
...
-System.import('angular2/platform/browser').then(function(browser_adapter) {
-  // TODO: once beta is out we should change this code to use a "test platform"
-  browser_adapter.BrowserDomAdapter.makeCurrent();
+Promise.all([
+  System.import('angular2/testing'),
+  System.import('angular2/platform/testing/browser'),
+]).then(function(imports) {
+  var testing = imports[0];
+  var browser = imports[1];
+  testing.setBaseTestProviders(browser.TEST_BROWSER_PLATFORM_PROVIDERS,
```
Comment (jelbourn) [inline test/karma-test-shim.js]: Could you add a comment explaining what this is doing?

File: tools/broccoli/broccoli-ts2dart.ts [hunk 3]
```diff
@@ -4,17 +4,22 @@
...
 import fse = require('fs-extra');
 import path = require('path');
-import * as ts2dart from 'ts2dart';
 import {wrapDiffingPlugin, DiffingBroccoliPlugin, DiffResult} from './diffing-broccoli-plugin';
 
 class TSToDartTranspiler implements DiffingBroccoliPlugin {
   static includeExtensions = ['.ts'];
 
-  private transpiler: ts2dart.Transpiler;
+  private transpiler: any /*ts2dart.Transpiler*/;
```
Comment (jelbourn) [inline tools/broccoli/broccoli-ts2dart.ts]: What happened to the types for ts2dart?

File: tools/broccoli/broccoli-ts2dart.ts [hunk 4]
```diff
@@ -4,17 +4,22 @@
...
+  private additionalFiles: string[];
 
   constructor(public inputPath: string, public cachePath: string,
-              public options: ts2dart.TranspilerOptions) {
+              public options: any /*ts2dart.TranspilerOptions*/) {
+    this.additionalFiles = options.additionalFiles || [];
+    delete options.additionalFiles;
```
Comment (jelbourn) [inline tools/broccoli/broccoli-ts2dart.ts]: Why does `additionalFiles` have to be deleted?

File: tools/broccoli/broccoli-ts2dart.ts [hunk 5]
```diff
@@ -22,16 +27,19 @@ class TSToDartTranspiler implements DiffingBroccoliPlugin {
     let toEmit = [path.resolve(this.inputPath, 'angular2/manual_typings/globals.d.ts')];
     let getDartFilePath = (path: string) => path.replace(/((\.js)|(\.ts))$/i, '.dart');
     treeDiff.addedPaths.concat(treeDiff.changedPaths)
-        .forEach((changedPath) => {
-          let inputFilePath = path.resolve(this.inputPath, changedPath);
+      .filter((name) => {
```
Comment (jelbourn) [inline tools/broccoli/broccoli-ts2dart.ts]: nit: don't need parens around `name` or `changedPath`, etc.

File: tools/broccoli/broccoli-ts2dart.ts [hunk 6]
```diff
@@ -4,17 +4,22 @@
...
   static includeExtensions = ['.ts'];
 
-  private transpiler: ts2dart.Transpiler;
+  private transpiler: any /*ts2dart.Transpiler*/;
+  private additionalFiles: string[];
 
   constructor(public inputPath: string, public cachePath: string,
-              public options: ts2dart.TranspilerOptions) {
+              public options: any /*ts2dart.TranspilerOptions*/) {
+    this.additionalFiles = options.additionalFiles || [];
```
Comment (jelbourn) [inline tools/broccoli/broccoli-ts2dart.ts]: Add a comment around what you _expect_ `additionalFiles` to be / what they're used for.

File: tools/broccoli/broccoli-ts2dart.ts (hunk 3 above)
Comment (hansl) [inline tools/broccoli/broccoli-ts2dart.ts]: https://github.com/angular/angular/commit/ae275fa4e469f0b30142260f06f2a16511dcd071#diff-cb068e9a6a1837586e67d28648bf2067L4

I'm not entirely sure. This is from the angular broccoli plugin.

File: test/karma-test-shim.js (hunk 2 above)
Comment (hansl) [inline test/karma-test-shim.js]: Done.

File: tools/broccoli/broccoli-ts2dart.ts (hunk 4 above)
Comment (hansl) [inline tools/broccoli/broccoli-ts2dart.ts]: Added a comment.

File: tools/broccoli/broccoli-ts2dart.ts (hunk 5 above)
Comment (hansl) [inline tools/broccoli/broccoli-ts2dart.ts]: Done.

File: tools/broccoli/broccoli-ts2dart.ts (hunk 6 above)
Comment (hansl) [inline tools/broccoli/broccoli-ts2dart.ts]: Done

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**